stream_options.go

```diff
@@ -0,0 +1,216 @@
+package eventsource
+
+import (
+	"net/http"
+	"time"
+)
+
+type streamOptions struct {
+	initialRetry        time.Duration
+	httpClient          *http.Client
+	lastEventID         string
+	logger              Logger
+	backoffMaxDelay     time.Duration
+	jitterRatio         float64
+	readTimeout         time.Duration
+	retryResetInterval  time.Duration
+	initialRetryTimeout time.Duration
+}
+
+// StreamOption is a common interface for optional configuration parameters that can be
+// used in creating a stream.
```

Comment:
Previously, StreamOption would modify the Stream struct itself. To avoid mixing configuration with implementation details, I changed it to operate on the streamOptions struct.